modules/imap/functions.php

```diff
@@ -1673,3 +1673,722 @@ function is_imap_archive_folder($server_id, $user_config, $current_folder) {
+        'RCPT command failed' => 'Failed to send email to APWG. The recipient address may be invalid or rejected by the SMTP server.',
+        'DATA command failed' => 'Failed to send email to APWG. The SMTP server did not accept the message data. Please try again later.',
+        '250' => 'Email was successfully sent to APWG (250 OK response received).',
+        '550' => 'Failed to send email to APWG. The recipient address was rejected by the mail server (550 error).',
+        '551' => 'Failed to send email to APWG. The recipient address does not exist (551 error).',
+        '552' => 'Failed to send email to APWG. The mail server rejected the message due to size limits (552 error).',
+        '553' => 'Failed to send email to APWG. The recipient address format is invalid (553 error).',
+        '554' => 'Failed to send email to APWG. The mail server rejected the message (554 error).'
+    );
+    
+    foreach ($error_mappings as $key => $message) {
+        if (strpos($error_msg, $key) !== false) {
+            return $message;
+        }
+    }
+    
+    return $error_msg;
+}}
+
+/**
+ * Create temporary file for spam report attachment
+ * @param string $sanitized_message The sanitized message content
+ * @param object $user_config User configuration object
+ * @param object $session Session object
+ * @param string $prefix File prefix (e.g., 'spamcop_' or 'apwg_')
+ * @return string Path to temporary file
+ */
+if (!hm_exists('create_spam_report_temp_file')) {
+function create_spam_report_temp_file($sanitized_message, $user_config, $session, $prefix) {
+    $file_dir = $user_config->get('attachment_dir', sys_get_temp_dir());
+    if (!is_dir($file_dir)) {
+        $file_dir = sys_get_temp_dir();
+    }
+
+    if ($file_dir !== sys_get_temp_dir() && $session) {
+        $user_dir = $file_dir . DIRECTORY_SEPARATOR . md5($session->get('username', 'default'));
+        if (!is_dir($user_dir)) {
+            @mkdir($user_dir, 0755, true);
+        }
+        $file_dir = $user_dir;
+    }
+    $temp_file = tempnam($file_dir, $prefix);
+    
+    if (class_exists('Hm_Crypt') && class_exists('Hm_Request_Key')) {
+        $encrypted_content = Hm_Crypt::ciphertext($sanitized_message, Hm_Request_Key::generate());
+        file_put_contents($temp_file, $encrypted_content);
+    } else {
+        file_put_contents($temp_file, $sanitized_message);
+    }
+    
+    return $temp_file;
+}}
+
+/**
+ * Fix encoding from 7bit to base64 for message/rfc822 attachment and extract boundary
+ * @param string $mime_message The MIME message
+ * @param string $service_name Service name for debug messages (e.g., 'SpamCop' or 'APWG')
+ * @return array Array with 'mime_message', 'mime_body', and 'boundary'
+ */
+if (!hm_exists('fix_spam_report_encoding')) {
+function fix_spam_report_encoding($mime_message, $service_name) {
```

Comment:
Can't we use a lib for this? I'd like to avoid all these low-lever regex operations if there is already a lib for this and we already are using mime manipulation libs + have a low-lever mime classes ourselves.